src/transformers/pipelines/fill_in_middle.py

```diff
@@ -0,0 +1,333 @@
+import enum
+import warnings
+from collections import OrderedDict
+
+from ..utils import add_end_docstrings, is_tf_available, is_torch_available
+from .base import PIPELINE_INIT_ARGS, Pipeline
+
+
+class ReturnType(enum.Enum):
+    TENSORS = 0
+    NEW_TEXT = 1
+    FULL_TEXT = 2
+
+
+if is_torch_available():
+    FIM_SUPPORTED_MODELS_TORCH = [
+        "codellama/CodeLlama-7b-hf",
+        "codellama/CodeLlama-13b-hf",
+        "codellama/CodeLlama-34b-hf",
+        "codellama/CodeLlama-7b-Python-hf",
+        "codellama/CodeLlama-13b-Python-hf",
+        "codellama/CodeLlama-34b-Python-hf",
+    ]
+
+
+@add_end_docstrings(PIPELINE_INIT_ARGS)
+class FIMPipeline(Pipeline):
+    """
+    Fill-in-the-middle Language generation pipeline. This pipeline predicts the middle part of a text given it's prefix and suffix
+    Example:
+    ```python
+    >>> from transformers import pipeline
+    >>> PROMPT = '''
+        def fibonacci(x: int) -> int:
+            <FILL_ME>
```

Comment:
down with this as that's what was done with CodeLLama! 
Let's make sure that the pipeline supports returning the filled code with the context in the ocrrect order!